api/v1/openapi.yaml

```diff
@@ -3013,3 +3022,86 @@ definitions:
     type: array
     items:
       "$ref": "#/definitions/Port"
+  EncryptionStatus:
+    description: |-
+      Status of transparent encryption
+
+      +k8s:deepcopy-gen=true
+    properties:
+      mode:
+        type: string
+        enum:
+          - Disabled
+          - IPsec
+          - Wireguard
+      msg:
+        type: string
+        description: Human readable status/error/warning message
+      wireguard:
+        description: Status of the Wireguard agent
+        "$ref": "#/definitions/WireguardStatus"
+  WireguardStatus:
+    description: |-
+      Status of the Wireguard agent
+
+      +k8s:deepcopy-gen=true
+    properties:
+      interfaces:
+        description: Wireguard interfaces managed by this Cilium instance
+        type: array
+        items:
+          "$ref": "#/definitions/WireguardInterface"
+  WireguardInterface:
+    description: |-
+      Status of a Wireguard interface
+
+      +k8s:deepcopy-gen=true
+    properties:
+      name:
+        description: Name of the interface
+        type: string
+      public-key:
+        description: Public key of this interface
+        type: string
+      listen-port:
+        description: Port on which the Wireguard endpoint is exposed
+        type: integer
+      peer-count:
+        description: Number of peers configured on this interface
+        type: integer
+      peers:
+        description: Optional list of wireguard peers
+        type: array
+        items:
+          "$ref": "#/definitions/WireguardPeer"
+  WireguardPeer:
+    description: |-
+      Status of a Wireguard peer
+
+      +k8s:deepcopy-gen=true
+    properties:
+      public-key:
+        description: Public key of this peer
+        type: string
+      endpoint:
+        description: Endpoint on which we are connected to this peer
+        type: string
+      last-handshake-time:
+        description: Endpoint on which we are connected to this peer
```

Comment:
I think the description is incorrect here

Reply:
Thanks, fixed! While at it, I also removed the "persistent-keep-alive" field from the proposed API, as we're currently not using it.